etcd/etcd2-live-http-to-https-migration.md

```diff
@@ -0,0 +1,192 @@
+
+```
+[Service]
+Environment="ETCD_LISTEN_CLIENT_URLS=http://0.0.0.0:2379,http://127.0.0.1:4001"
+```
+
+Then run `systemctl daemon-reload` and `systemctl restart etcd2` to restart etcd2. Check etcd2 availability using these commands:
+
+```sh
+etcdctl member list
+etcdctl cluster-health
+```
+
+Repeat these steps on each CoreOS host which has etcd2 member.
+
+### Generate TLS keypairs
+
+Follow [generate self signed certificates][self-signed-ca] doc to generate self-signed certificates.
+
+### Copy Keypairs
+
+Let's assume we have three CoreOS hosts with three etcd2 cluster members: server1, server2, server3 and corresponging IP addresses: 172.16.0.101, 172.16.0.102, 172.16.0.103. We will use following keypairs filenames in our example:
+
+```
+server1.pem
+server1-key.pem
+```
+
+Copy your `ca.pem` CA certificate into `/etc/ssl/certs` directory and run `update-ca-certificates` script to update your certificates bundle.
+
+Create `/etc/etcd2` directory, copy there corresponding certificate and key and set file permissions:
+
+```
+chown -R etcd:etcd /etc/etcd2/
+chmod 600 /etc/etcd2/*-key.pem
+```
+
+Repeat this step on the rest member nodes.
+
+### Configure etcd2 Proxy
+
+In case when you use etcd2 proxy you have to generate client keypair (i.e. `client1.pem` and `client1-key.pem`) and follow previous step.
+
+## Configure etcd2 Keypair
+
+Now we should configure keypair path's for etcd2. Create `/etc/systemd/system/etcd2.service.d/30-certs.conf` [drop-in][drop-ins] with contents below:
+
+```
+[Service]
+Environment="ETCD_CERT_FILE=/etc/etcd2/server1.pem"
+Environment="ETCD_KEY_FILE=/etc/etcd2/server1-key.pem"
+Environment="ETCD_PEER_CERT_FILE=/etc/etcd2/server1.pem"
+Environment="ETCD_PEER_KEY_FILE=/etc/etcd2/server1-key.pem"
+```
+
+Reload systemd config files `systemctl daemon-reload` and restart etcd2 `systemctl restart etcd2`. Then check your cluster health:
+
+```sh
+etcdctl member list
+etcdctl cluster-health
+```
+
+If everything is ok, repeat this step on the rest member nodes.
+
+### Configure etcd2 Proxy
+
+In case when you use etcd2 proxy you have to use `/etc/systemd/system/etcd2.service.d/30-certs.conf` [drop-in][drop-ins] with the contents below:
+
+```
+[Service]
+Environment="ETCD_CERT_FILE=/etc/etcd2/client1.pem"
+Environment="ETCD_KEY_FILE=/etc/etcd2/client1-key.pem"
+# listen only loopback interface for security reasons
+Environment="ETCD_LISTEN_CLIENT_URLS=http://127.0.0.1:2379,http://127.0.0.1:4001"
+```
+
+Reload systemd config files `systemctl daemon-reload` and restart etcd2 `systemctl restart etcd2`. Then check proxy health: 
+
+```sh
+curl http://127.0.0.1:4001/v2/stats/self
+```
+
+## Change etcd2 Peer URLs
+
+Command below should generate commands which you have to run manually on one of the member node:
```

Comment:
Have to add note about:
etcdctl member list and describe which fields and where should we use